### DISCUSSION TITLE
XPath2 and text()

### DISCUSSION THREAD

**Discussion by bassmonger** [appium/appium#17147] [General]
Hello all - I've had a search for this and cant see it mentioned anywhere, apologies if it's been covered...

In older versions of Appium (with xpath1), I used to use xpaths such as this: 

`//*[contains(text(), 'text') or contains(@text, 'text')]
`

This let me not worry whether we were in a web or native app context, it'd just work either way. However, in the latest version (using xpath2), the following will work on a native app:

`//*[matches(@text, 'text')]
`

...but this won't:

`//*[matches(text(), 'text')]
`

... and gives the usual "org.eclipse.wst.xml.xpath2.processor.DynamicError: Value does not match a required type" exception that you get when using 'contains()' in an xpath1-ish way. I don't think this is by design though - surely text() should be a valid selector?

Any ideas? Is it a bug in org.eclipse.wst.xml.xpath2... and/or do I need to either tailor my xpaths depending on context (or use enforceXPath1 = true)?

**Comment by Mr-Ples** [comment]
Depending on the driver you are using only XPATH 1 might be supported (as is the case with XCUITest driver: https://github.com/appium/appium-xcuitest-driver)

> Only Xpath 1.0 is supported.

Also xpath is slowest locator strategy (on ios at least) https://github.com/facebookarchive/WebDriverAgent/wiki/How-To-Achieve-The-Best-Lookup-Performance
